### DISCUSSION TITLE
Setting up ready for account to be activated

### DISCUSSION THREAD

**Discussion by deltamelter** [BottlecapDave/HomeAssistant-OctopusEnergy#277] [Q&A]
I only signed up to Octopus Agile on Saturday so my account is not fully live yet but I was hoping to get a head start on controlling my equipment based on agile rates.
The API is already active and getting list of previous, current and future 30 minute rates, the meters are present (I have mpan and mprn) but so far no readings (I don't switch until later in the week)
The following API URL gives me the the half hourly rates
E-1R-AGILE-FLEX-22-11-25-{region_code}
I installed this plugin and only get 2 sensors:
binary_sensor.octopus_energy_saving_sessions
sensor.octopus_energy_saving_session_points
I just tried to follow the setup through UI but I get presented with this form where I assume the mpan would go but it isn't an editable box:
![Screenshot 2023-06-19 112702](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/27061628/8659ea6c-de29-4ee6-a0d5-7e5bd337b099)
What is this relying on, any way to complete setup before the service goes live?

**Comment by BottlecapDave** [comment]
Hi there and thanks for using the integration.

Currently the integration requires you to have an active agreement attached to a meter for it's sensors to appear, which I imagine is why you only have two sensors. This is because some people have their meters replaced which still come back in the data and therefore need a way of filtering out the data. I'll look at making this more lenient with future agreements, but it won't make an appearance before you account becomes active.

**Comment by deltamelter** [comment]
OK, makes sense.  Didn't realise this all was dependent on the agreements which still show the initial VAR tariff code until activation.  I notice that the agreement also has a future agreement date, so I guess I'll just wait :smile: 
I was thinking that things like agile half-hourly current_rates sensor should be available because that is taken from a generic URL, but I suppose it takes the tariff_code and region_code from agreements.  Is there a way to "hard code" these so I can use the "rate card" already, or should I just stick to google sheets graph for a few more days to satisfy my curiosity :nerd_face:  ?

**Reply by BottlecapDave** [reply]
The only way would be to edit the code to return a fixed agreement shape, but tbh I'd just try to wait until your agreement comes into effect. I can understand the frustration :) Once it does, you will need to reload the integration to get the meters to show.
